parsers/javascript/index.js

```diff
@@ -0,0 +1,34 @@
+import { parseLine } from "./utils.js";
+
+export const parseAtOnce = (input) => {
+  return input.split("\n").map(parseLine);
+}
+
+export default () => {
+  let buffer = "";
+
+  const parseLinesFromBuffer = (controller) => {
+    while (buffer.includes("\n")) {
+      const indexOfNewLine = buffer.indexOf("\n");
+      const line = buffer.substring(0, indexOfNewLine);
+      const parsedBlock = parseLine(line);
+      controller.enqueue(parsedBlock);
+      buffer = buffer.substring(indexOfNewLine + 1);
+    }
+  }
+
+  return new TransformStream({
```

Comment:
What if we expressed the streaming parser in terms of a generator function instead of the WebStreams API? A possible benefit I see to taking that approach is nice ergonomics, thanks to language support such as `for await`.

Any reflections on why streams vs generators?

Reply:
I have played around with generator functions but could not find an ergonomic way to use them. As far as I understand, you can only pass arguments when calling generator functions initially, but not later on when calling `Function.next()`. But my experience is limited.

I chose Web Streams in the first place because I can simply pipe data from other APIs (e. g. fetch API or File System Access API) through the parser that way.